doc/sphinx_source/tutorials/module.rst

```diff
@@ -0,0 +1,234 @@
+
+There are a few required functions a module must perform in order to properly work with Eggdrop
+
+Module Header
+-------------
+
+A module should include license information. This tells other open source users how they are allowed to use the code. Eggdrop uses GPL 2.0 licensing, and our license information looks like this::
+
+  /*
+  * This program is free software; you can redistribute it and/or
+  * modify it under the terms of the GNU General Public License
+  * as published by the Free Software Foundation; either version 2
+  * of the License, or (at your option) any later version.
+  *
+  * This program is distributed in the hope that it will be useful,
+  * but WITHOUT ANY WARRANTY; without even the implied warranty of
+  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  * GNU General Public License for more details.
+  *
+  * You should have received a copy of the GNU General Public License
+  * along with this program; if not, write to the Free Software
+  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+  */
+
+Required Code
+-------------
+
+For this section, you don't necessarily need to understand what it is doing, but this code is required for a module to function.
+
+You'll next want to name your module::
+
+  #define MODULE_NAME "woobie"
+
+Declare your own function tables (you don't need to understand this part; you just need to copy/paste it)::
+
+  #undef global
+  static Function *global = NULL, *server_funcs = NULL;
+  EXPORT_SCOPE char *woobie_start();
+
+Next are two memory-reporting functions used by the core Eggdrop .status and .module commands::
+
+  static int woobie_expmem()
+  {
+    int size = 0;
+
+    return size;
+  }
+
+  static void woobie_report(int idx, int details)
+  {
+    if (details) {
+      int size = woobie_expmem();
+
+      dprintf(idx, "    Using %d byte%s of memory\n", size,
+              (size != 1) ? "s" : "");
+    }
+  }
+
+This function is calld when Eggdrop loads the module::
+
+  char *woobie_start(Function *global_funcs)
+  {
+    global = global_funcs;
+
+    /* Register the module. */
+    module_register(MODULE_NAME, woobie_table, 2, 1);
+    /*                                            ^--- minor module version
+     *                                         ^------ major module version
+     *                           ^-------------------- module function table
+     *              ^--------------------------------- module name
+     */
+
+    if (!module_depend(MODULE_NAME, "eggdrop", 108, 0)) {
```

Comment:
do you want to explain why 108 means 1.8.0? I don't get it

Reply:
"Because". That's how the code is done throughout, I assume to account for not having a patchlevel check. ((major*100) + minor)